stl/inc/ranges

```diff
@@ -983,13 +983,21 @@ namespace ranges {
         using iterator_category = input_iterator_tag;
     };
 
-    // clang-format off
     template <weakly_incrementable _Wi>
         requires copyable<_Wi>
-    struct _Ioterator : _Ioterator_category_base<_Wi> {
-        // clang-format on
+    class _Ioterator : public _Ioterator_category_base<_Wi> {

```

Comment:
What are we gaining from making it a class?

Reply:
Nothing - this is an unrelated change to follow our "classes with no privates are structs" convention. I'll revert this to keep the changes in the LLVM update focused, and do it in a followup.